pkg/controller/nodes/task/backoff/handler.go

```diff
@@ -0,0 +1,219 @@
+			}
+			// It is necessary to parse the error message to get the actual constraints
+			// in this case, if the error message indicates constraints on memory only, then we shouldn't be used to lower the CPU ceiling
+			// even if CPU appears in requestedResourceList
+			newCeiling := GetComputeResourceAndQuantityRequested(err)
+			h.ComputeResourceCeilings.updateAll(&newCeiling)
+
+			return errors.Wrapf(errors.BackOffError, err, "The operation was attempted but failed")
+		}
+		logger.Infof(ctx, "The operation was attempted but failed due to reason(s) other than insufficient resource: [%v]\n", err)
+		return err
+	} else { // The backoff is active and the resource request exceeds the ceiling
+		logger.Infof(ctx, "The operation was blocked due to back-off")
+		return errors.Errorf(errors.BackOffError, "The operation attempt was blocked by back-off "+
+			"[attempted at: %v][the block expires at: %v] and the requested "+
+			"resource(s) exceeds resource ceiling(s)", now, h.SimpleBackOffBlocker.getBlockExpirationTime())
+	}
+
+}
+
+func IsResourceQuotaExceeded(err error) bool {
+	return errors2.IsForbidden(err) && strings.Contains(err.Error(), "exceeded quota")
+}
+
+func GetComputeResourceAndQuantityRequested(err error) v1.ResourceList {
+	// re := regexp.MustCompile(`(?P<part>(?P<key>requested|used|limited): limits.(?P<resource_type>[a-zA-Z]+)=(?P<quantity_expr>[a-zA-Z0-9]+))`)
+	// Playground: https://play.golang.org/p/oOr6CMmW7IE
+	// limits.cpu=7,limits.memory=64Gi, used: limits.cpu=249,limits.memory=2012730Mi, limited: limits.cpu=250,limits.memory=2000Gi
+	// re := regexp.MustCompile(`(?P<key>requested): (limits.(?P<resource_type>[a-zA-Z]+)=(?P<quantity_expr>[a-zA-Z0-9]+)[,]*)+`)
+	// re := regexp.MustCompile(`(?P<key>requested): (limits.(?P<resource_type>[a-zA-Z]+)=(?P<quantity_expr>[a-zA-Z0-9]+)[,]*)+`)
+
+	// Extracting "requested: limits.cpu=7,limits.memory=64Gi"
+	re := regexp.MustCompile(`requested: (limits.[a-zA-Z]+=[a-zA-Z0-9]+[,]*)+`)
+	matches := re.FindAllStringSubmatch(err.Error(), -1)
+	requestedComputeResources := v1.ResourceList{}
+
+	if len(matches) == 0 || len(matches[0]) == 0 {
+		return requestedComputeResources
+	}
+
+	// Extracting "limits.cpu=7,limits.memory=64Gi"
+	descr := strings.SplitN(matches[0][0], ":", 2)
+
+	// Extracting "limits.cpu=7","limits.memory=64Gi"
+	chunks := strings.SplitN(descr[1], ",", -1)
+	for _, c := range chunks {
+		// Extracting "cpu=7","memory=64Gi"
+		resrcString := strings.SplitN(c, ".", 2)
+		if len(resrcString) < 2 {
+			continue
+		}
+		// Extracting ["cpu","7"], ["memory","64Gi"]
+		tuple := strings.SplitN(resrcString[1], "=", 2)
+		if len(tuple) < 2 {
+			continue
+		}
+		requestedComputeResources[v1.ResourceName(tuple[0])] = resource.MustParse(tuple[1])
+	}
+	return requestedComputeResources
+}
+
+// TODO ssingh: clean it and move to its right place
+func IsBackoffError(err error) bool {
+	code, found := errors3.GetErrorCode(err)
```

Comment:
I think you can use `errors3.IsErrorCode(err, errors.BackOffError)`
and maybe rename `errors3` to `stdErrs` or something